terraform/modules/fake_server_resources/fake_server_resources.tf

```diff
@@ -60,30 +104,185 @@ resource "google_storage_bucket_object" "portal_server_global_manifest" {
   })
 }
 
-# Constructs global manifests for this env's ingestion servers, populated with
-# public keys that match what the sample_maker will use.
-resource "google_storage_bucket_object" "ingestor_global_manifests" {
-  for_each     = toset(var.ingestors)
-  name         = "${each.key}/global-manifest.json"
-  bucket       = var.manifest_bucket
-  content_type = "application/json"
-  content = jsonencode({
-    format = 1
-    server-identity = {
-      # these values are ignored in our test setup; see data_share_processor.tf
-      aws-iam-entity            = "irrelevant in test environment"
-      gcp-service-account-id    = "0"
-      gcp-service-account-email = "irrelevant in test environment"
+resource "kubernetes_namespace" "tester" {
+  metadata {
+    name = "tester"
+    annotations = {
+      environment = var.environment
     }
-    batch-signing-public-keys = {
-      # This key identifier matches the one passed to sample_maker's
-      # --batch-signing-private-key-identifier parameter in kubernetes.tf
-      sample-maker-signing-key = {
-        # This public key matches the one passed to sample_maker's
-        # --batch-signing-private-key parameter in kubernetes.tf
-        public-key = "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAENpmX5uFAu9z5FrGM91Lm+lEyABUA\njxsaLj9TtT5Zp7TqXy2Hb8mQwrnpwM79aAn8k6KJTQKzY8KP/oWqzZ9X7A==\n-----END PUBLIC KEY-----"
-        expiration = "some date"
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_iam_role" "tester_role" {
+  name = "prio-${var.environment}-integration-tester"
+  # Since azp is set in the auth token Google generates, we must check oaud in
+  # the role assumption policy, and the value must match what we request when
+  # requesting tokens from the GKE metadata service in
+  # S3Transport::new_with_client
+  # https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_iam-condition-keys.html
+  assume_role_policy = <<ROLE
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "accounts.google.com"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "accounts.google.com:sub": "${module.account_mapping.google_service_account_unique_id}",
+          "accounts.google.com:oaud": "sts.amazonaws.com/${data.aws_caller_identity.current.account_id}"
+        }
       }
     }
-  })
+  ]
+}
+ROLE
+
+  tags = {
+    environment = "prio-${var.environment}"
+  }
+}
+
+resource "aws_iam_role_policy" "bucket_role_policy" {
+  name = "prio-${var.environment}-integration-tester-policy"
+  role = aws_iam_role.tester_role.id
+
+  policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:*"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+POLICY
+}
```

Comment:
The resources `aws_iam_role_policy.bucket_role_policy` and `aws_iam_role.tester_role` duplicate a lot of code (down to the comments) from `data_share_processor.tf`. Perhaps it could be factored out into a small module?

Reply:
I'll take a look into doing this!